_moderngl.py

```diff
@@ -366,5 +366,168 @@ def make_varying(name, number, array_length, dimension):
     return res
 
 
+def parse_spv_inputs(raw_spv: bytes):
+    spv = struct.unpack(str(len(raw_spv)//4)+'I', raw_spv)
```

Comment:
do not parse everything. make it lazy by using the lambda i previously suggested.

for your example it might be a bit slower, but for actual code where the type information is a small part compared to the computation logic it would matter more.

Reply:
Perhaps you're right, I've already tried lambda recently, but I didn't like the speed then. I'll put it on again.